spec/support/acceptance/shared_context.rb

```diff
@@ -6,22 +6,31 @@
                splunk_token_manifest
              end
 
-  before(:all) do
+  before(:each) do
     master.apply_manifest(manifest, catch_failures: true)
-    5.times do
-      response = master.run_shell("puppet task run enterprise_tasks::test_connect -n #{master.uri}")
-      raise 'task run failed' unless response.exit_code == 0
-    end
-    # 'Waiting for the cron job to complete'
-    @time_stamp ||= Time.now
+  end
+end
+
+RSpec.shared_context 'send and collect events' do
+  before(:each) do
+    cmd = %(for i in {1..#{TASK_COUNT}}; do puppet task run enterprise_tasks::test_connect -n #{master.uri}; done)
```

Comment:
btw, I just testing this line manually by replacing `TASK_COUNT` with a value that was too high and then a value too low and in either case the test failed as expected.